sql/item_jsonfunc.cc

```diff
@@ -1150,6 +1150,76 @@ static int check_contains(json_engine_t *js, json_engine_t *value)
 }
 
 
+longlong Item_func_json_equals::val_int()
+{
+  String *js1= args[0]->val_json(&tmp_js);
+  String *js2;
+  String *val1, *val2;
+
+  json_engine_t je1, je2, ve1, ve2;
+
+  if (args[0]->null_value)
+    goto return_null;
```

Comment:
Your logic looks good, but can we do this without using `goto`?
Maybe by using a separate function, and returning an error code?

Reply:
"goto" has the advantage of producing less code (if the compiler is not smart enough to optimize the function call in the first place).

Although I am not personally a fan of goto statements, the error handling at the end, as is now, seems to be the cleanest form of writing this.